packages/content/src/types.ts

```diff
@@ -0,0 +1,137 @@
+import { IItem, IGroup } from "@esri/arcgis-rest-types";
+
+export type HubType = 'document' | 'dataset' | 'map' | 'app' | 'site' | 'initiative' | 'event' | 'template';
+
+// Based on the [Hub Content UI Inventory](https://app.lucidchart.com/invitations/accept/dab82a25-2ed9-4f63-8075-3db8873087e3)
+// Attempting a simple _Hungarian Notation_ for attributes suffix: `*Date`, `*Id`, `*Number`. Plural (e.g. `tags`) is an array.
+
+export interface IContent extends IItem {
+  
+  // Inherited metadata (noting relevant subset of all item attributes)
+  // id: string; // item.id
+  // title: string; // item.title
+  // snippet: string; // item.snippet
+  // description: string; // item.description
+  // categories?: Array<string>; // item.categories
+  // tags: Array<string>; // item.tags  
+  // culture: string; // written locale e.g. en, es, cn
+  // url?: string;
+  // size: number;
+
+  // Derived metadata
```

Comment:
If we think about the `update` flow for an `IContent`, and everything below this point are "derived" properties... maybe we should mark all of them as `readonly`? 
https://www.typescriptlang.org/docs/handbook/interfaces.html#readonly-properties

Reply:
interesting idea. Along those same lines, I've been tempted to think about using an [abstract class](https://www.typescriptlang.org/docs/handbook/classes.html#abstract-classes) so we could actually implement things like `createdDate` etc as getters. I know that's a whole other can of worms b/c we get into the instance vs POJO debate.

Reply:
@dbouwman Generally "derived" can be bi-directional. So we could set `content.metadata.contactInfo.address` and have that go into the `content/items/:id/metadata` endpoint.

Reply:
Many of the derived properties are not bi-directional, and ideally we'd want to keep them from being overwritten and diverging from their source. TS lures you into thinking you can actually do that, but at the end of the day JS consumers can do `content.createdDate = 'ha ha!'`, so we should be judicious about how much effort we put into trying to control this. Seems to me that flagging the ones we intend to not be bi-directional w/ `readonly` is a reasonable start.